Mani-Quotify/app/src/main/java/com/mani/quotify007/ui/screens/quote/QuotesScreen.kt

```diff
@@ -65,16 +74,40 @@ fun QuotesScreen(quote: Quote, onEvent: (MainEvent) -> Unit) {
             modifier = Modifier.fillMaxWidth(),
             horizontalArrangement = Arrangement.SpaceEvenly
         ) {
-            TextButton(onClick = { onEvent(MainEvent.AddFavorite(quote)) }) {
-                Icon(imageVector = Icons.Default.Favorite, contentDescription = "Save")
+            TextButton(onClick = {
+                if (isAddOnly) {
+                    if (!quote.isFavorite) {
+                        quote.isFavorite = true
+                        onEvent(MainEvent.AddFavorite(quote))
+                    } else {
+                        Toast.makeText(context, "Already added to favorites", Toast.LENGTH_SHORT).show()
```

Comment:
can we shift context based thing to activity?

Reply:
Resolved